src/cloudformation_cli_python_lib/interface.py

```diff
@@ -59,7 +59,31 @@ class HandlerErrorCode(str, _AutoName):
 
 class BaseResourceModel:
     def _serialize(self) -> Mapping[str, Any]:
-        return self.__dict__
+        ser = self.__dict__
+        del_keys = []
+        for k, v in ser.items():
+            if isinstance(v, list):
+                ser[k] = self._serialize_list(v)
+            elif isinstance(v, BaseResourceModel):
+                ser[k] = v._serialize()  # pylint: disable=protected-access
+            elif v is None:
+                del_keys.append(k)
+            else:
+                ser[k] = v
+        for k in del_keys:
+            del ser[k]
+        return ser
+
+    def _serialize_list(self, v: List[Any]) -> List[Any]:
```

Comment:
can be static?

Reply:
I left this as is, as it's now using `self._serialize_item`, could have made it point to `BaseResourceModel._serialize_item` but that felt a bit unnatural and will require a pylint override for accessing the private method.